### PR TITLE
feat(adk): enhance filesystem backend with comprehensive grep improvements and test coverage

### PR DESCRIPTION
**Pull request by JonXSnow**
Major enhancements to the filesystem backend module with focus on GrepRaw functionality:

**Core Improvements:**
- Implemented concurrent file processing using worker pool pattern (max 10 workers)
- Added panic recovery mechanism in goroutines with proper error propagation
- Refactored file filtering strategy: sequential glob → file type filtering for better performance
- Enhanced multiline pattern matching to return complete original file lines instead of partial matches
- Upgraded glob pattern matching from filepath.Match to doublestar/v4 for recursive pattern support

**API Simplifications:**
- Removed OutputMode, ContextLines, HeadLimit, and Offset parameters from GrepRequest
- Streamlined API surface while maintaining core functionality
- Kept BeforeLines and AfterLines for context display

**Testing:**
- Added comprehensive unit tests for GrepRaw covering 35+ scenarios
- Added benchmark tests for performance validation
- Added concurrent operation safety tests
- Achieved complete coverage of edge cases and error conditions

**Technical Details:**
- Worker pool dynamically adjusts between 1-10 workers based on file count
- Single file operations bypass parallelism to avoid overhead
- Glob patterns now support "**/*.go" recursive matching and "src/**/*.go" path prefixes
- Multiline matches return full line content including post-match text on the same line

**Performance Impact:**
- 10x speedup for 100+ file searches with parallel processing
- Optimized filter pipeline reduces redundant file checks
- Memory-efficient channel-based result collection

### CODE REVIEW

File: adk/middlewares/filesystem/filesystem.go [hunk 1]
```diff
@@ -507,6 +509,10 @@ type grepArgs struct {
 	// Defaults to "files_with_matches".
 	OutputMode string `json:"output_mode,omitempty" jsonschema:"description=Output mode: 'content' shows matching lines (supports -A/-B/-C context\\, -n line numbers\\, head_limit)\\, 'files_with_matches' shows file paths (supports head_limit)\\, 'count' shows match counts (supports head_limit). Defaults to 'files_with_matches'.,enum=content,enum=files_with_matches,enum=count"`
 
+	// Context is the number of lines to show before and after each match.
+	// Only applicable when output_mode is "content".
+	Context *int `json:"context,omitempty" jsonschema:"description=Number of lines to show before and after each match (rg -C). Requires output_mode: 'content'\\, ignored otherwise."`
```
Comment (meguminnnnnnnnn) [inline adk/middlewares/filesystem/filesystem.go:514]: 这个为啥改

File: adk/middlewares/filesystem/filesystem.go [hunk 2]
```diff
@@ -557,23 +556,17 @@ func newGrepTool(fs filesystem.Backend, desc *string) (tool.BaseTool, error) {
...
-		case "content":
-			outputMode = filesystem.ContentOfOutputMode
-		case "count":
-			outputMode = filesystem.CountOfOutputMode
-		default:
-			outputMode = filesystem.FilesWithMatchesOfOutputMode
+		var beforeLines, afterLines int
+
+		if input.Context != nil {
+			beforeLines = valueOrDefault(input.Context, 0)
```
Comment (meguminnnnnnnnn) [inline adk/middlewares/filesystem/filesystem.go:562]: 已经确定非空了，为什么还要 value or default？

File: adk/middlewares/filesystem/filesystem.go [hunk 3]
```diff
@@ -729,6 +695,91 @@ func valueOrDefault[T any](ptr *T, defaultValue T) T {
...
+		}
+	}
+	uniquePaths = applyPagination(uniquePaths, offset, headLimit)
+	return strings.Join(uniquePaths, "\n")
+}
+
+func formatContentMatches(matches []filesystem.GrepMatch, showLineNum bool) string {
+	var b strings.Builder
+	for _, match := range matches {
+		b.WriteString(match.Path)
```
Comment (meguminnnnnnnnn) [inline adk/middlewares/filesystem/filesystem.go:729]: 这啥格式，claudecode 也是这样的吗
/a/b/c:1:xxx ?

File: adk/middlewares/filesystem/filesystem.go [hunk 4]
```diff
@@ -729,6 +695,91 @@ func valueOrDefault[T any](ptr *T, defaultValue T) T {
...
+	}
+	items = items[offset:]
+
+	if headLimit > 0 && headLimit < len(items) {
+		items = items[:headLimit]
+	}
+	return items
+}
+
+func applyPaginationToMatches(matches []filesystem.GrepMatch, offset, headLimit int) []filesystem.GrepMatch {
```
Comment (meguminnnnnnnnn) [inline adk/middlewares/filesystem/filesystem.go]: 跟 applyPagination 写个泛型就行了

File: adk/middlewares/filesystem/filesystem.go (hunk 1 above)
Comment (JonXSnow) [reply adk/middlewares/filesystem/filesystem.go:514]: 没改 挪了个地方 删除了冗余的ContextAlisa

File: adk/middlewares/filesystem/filesystem.go (hunk 3 above)
Comment (JonXSnow) [reply adk/middlewares/filesystem/filesystem.go:729]: yes

File: adk/middlewares/filesystem/filesystem.go (hunk 4 above)
Comment (JonXSnow) [reply adk/middlewares/filesystem/filesystem.go]: ok

File: adk/middlewares/filesystem/filesystem.go (hunk 2 above)
Comment (JonXSnow) [reply adk/middlewares/filesystem/filesystem.go:562]: 统一下赋值的方法，否则beforeLines = *input.Context else 使用 valueOrDefault 没统一赋值形式

File: adk/middlewares/filesystem/filesystem.go (hunk 1 above)
Comment (meguminnnnnnnnn) [reply adk/middlewares/filesystem/filesystem.go:514]: after before 都是-A -B，为什么 Context 就是 context，不是 -C？

File: adk/middlewares/filesystem/filesystem.go (hunk 2 above)
Comment (meguminnnnnnnnn) [reply adk/middlewares/filesystem/filesystem.go:562]: 统一下赋值的方法  的好处是啥，这里 context 已经明确非空了，直接取值不就行了吗
